server/routes.go

```diff
@@ -64,22 +64,22 @@ var loaded struct {
 var defaultSessionDuration = 5 * time.Minute
 
 // load a model into memory if it is not already loaded, it is up to the caller to lock loaded.mu before calling this function
-func load(c *gin.Context, modelName string, reqOpts map[string]interface{}, sessionDuration time.Duration) (*Model, error) {
+func load(c *gin.Context, modelName string, reqOpts map[string]interface{}, sessionDuration time.Duration) (*Model, *api.Options, error) {
```

Comment:
What might be a bit cleaner here (vs passing in and returning `api.Options`) is to determine the options ahead of time, them pass that in here. That stuff can be done in the function that calls `load`

It's a larger chance, but this way `load` won't be responsible for determining the final options, instead it will just load the model with the options provided

Reply:
I was debating doing this too, it wasn't much work so I made the change